hpcrs/docs/hpcrs_tutorial.md

```diff
@@ -0,0 +1,254 @@
+3. Install helm (v3.3.4) .  Follow the instructions at
+https://www.techrepublic.com/article/how-to-install-the-kubernetes-package-manager-helm/ 
+
+4. Setup the local registry for the cri image  
+
+5. Setup the local registry on the master node, and download the HPCRS stack image:  
+```
+#docker pull docker.io/registry 
+#docker run -d -p 5000:5000 --name=registry --restart=always --privileged=true  --log-driver=none -v /registery:/tmp/registry registry 
+#docker pull <your registry address or url>/sysstacks/hpc_icc:v0.1.0-rc3
+```
+
+6. Modify the Dockerfile to add QE based on the HPCRS stack image  
+`# vi Dockerfile
+From <your registry url>/sysstacks/hpc_icc:v0.1.0-rc3`
+
+## Add and Build QE
+1. For details refer to: https://hpc-forge.cineca.it/files/gara_tier_1/public/Benchmark-Instructions.txt. A refrence `make.inc` file is provided in the image folder 
+```
+#Proxy Settings
+ARG proxy=<your proxy if needed>
+ENV http_proxy=$proxy  
+ENV https_proxy=$proxy  
+#Install ssh and Generate ssh Host Keys
+RUN swupd bundle-add clr-network-troubleshooter
+RUN swupd bundle-add openssh-server
+COPY ssh-entrypoint.sh /bin/ssh-entrypoint.sh
+RUN chmod +x /bin/ssh-entrypoint.sh
+RUN ssh-entrypoint.sh
+#Fix ssh Login Issue
+RUN usermod -p ! root
+RUN mkdir -p /etc/ssh/
+RUN echo "Port 2022" > /etc/ssh/sshd_config
+```
+
+2. Create the cri image based on HPCRS in the registry  
+```
+  #cd image  
+   #podman build -t hpcrstestcri:cri --build-arg proxy=$http_proxy -f Dockerfile . 
+   #podman push hpcrstestcri:cri --tls-verify=false docker://localhost:5000/hpcrstestcri:cri
+```   
+
+3. Copy the config to the root user directory and install flannel  
+```
+   #mkdir -p $HOME/.kube && cp /etc/kubernetes/admin.conf $HOME/.kube/config  
+   #kubectl apply -f https://raw.githubusercontent.com/coreos/flannel/master/Documentation/kube-flannel.yml
+   ```
+
+4. Init the Kubernetes cluster  
+```
+  #kubeadm reset --cri-socket=/run/crio/crio.sock -f 
+  #kubeadm init --apiserver-advertise-address <x.x.x.x> --pod-network-cidr <x.x.x.x>/16 --cri-socket=/run/crio/crio.sock --ignore-preflight-errors=SystemVerification --token-ttl=0
+  ```
+
+5. Record the last command in the log, which will be used to join workers to the Kubernetes cluster, such as:  
+```
+kubeadm init --apiserver-advertise-address <your network address> --pod-network-cidr <your network address>/16 --cri-socket=/run/crio/crio.sock --ignore-preflight-errors=SystemVerification --token-ttl=0
+```
+ 
+6. Run the following commands on each worker to join the cluster:  
+```
+#kubeadm join <your network address>:6443 --token tceoji.65vxsvcv8z5vzce2 --discovery-token-ca-cert-hash sha256:d4b3848ff2d5a242b63037687351805abb7abecc8a260e77905ca61d9d077758 
+``` 
+***NOTE***  
+Make sure the time is aligned across worker and master nodes, checking with the Linux  `#date` command.
+ 
+ 
+## Run  QE on the HPCRS image
+All commands are run on the Kubernetes master node.  
+
+  ```
+  #export NAMESPACE=helm  
```

Comment:
A lot of LSEPs made it into here

Reply:
Bad markup editor, fixed in latest commit.